File: Data Structures & Algorithms/two-integer-sum/submission-1.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        di = {}
        for idx in range(len(nums)):
            if nums[idx] in di:
                return [di[nums[idx]], idx]
            di[target-nums[idx]] = idx
    

File: Data Structures & Algorithms/binary-search/submission-1.rs
impl Solution {
    pub fn search(nums: Vec<i32>, target: i32) -> i32 {
        let (mut ls, mut rs) = (0i32, nums.len() as i32 -1);
        while ls <= rs {
            let mid = ls + (rs-ls)/2;
            if nums[mid as usize] > target {
                rs = mid -1;
            } else if nums[mid as usize] < target {
                ls = mid +1;
            } else {
                return mid;
            }
        }
        -1
    }
}
